extensions/lifecycle/LifecycleMetrics.js

```diff
@@ -0,0 +1,89 @@
+const { ZenkoMetrics } = require('arsenal').metrics;
+
+const LIFECYCLE_LABEL_ORIGIN =  'origin';
+const LIFECYCLE_LABEL_PROCESS = 'process';
+const LIFECYCLE_LABEL_OP = 'op';
+const LIFECYCLE_LABEL_STATUS = 'status';
+
+const conductorLastBatchTime = ZenkoMetrics.createGauge({
+    name: 'lifecycle_conductor_last_batch_time',
+    help: 'Timestamp of latest lifecycle batch',
+    labelNames: [
+        LIFECYCLE_LABEL_ORIGIN
+    ],
+});
+
+const lifecycleS3Operations = ZenkoMetrics.createCounter({
+    name: 'lifecycle_s3_operations',
+    help: 'Total number of S3 operations by the lifecycle processes',
+    labelNames: [
+        LIFECYCLE_LABEL_ORIGIN,
+        LIFECYCLE_LABEL_OP,
+        LIFECYCLE_LABEL_STATUS,
+        LIFECYCLE_LABEL_PROCESS,
+    ],
+});
+
+const lifecycleKafkaPublish = ZenkoMetrics.createCounter({
+    name: 'lifecycle_kafka_publish',
+    help: 'Total number messages published by lifecycle processes',
+    labelNames: [
+        LIFECYCLE_LABEL_ORIGIN,
+        LIFECYCLE_LABEL_STATUS,
+        LIFECYCLE_LABEL_OP,
+        LIFECYCLE_LABEL_PROCESS,
+    ],
+});
+
+function getStatusLabel(statusCode) {
+    if (!statusCode) {
+        return '2xx';
+    }
+
+    if (statusCode >= 500) {
+        return '5xx';
+    }
+    if (statusCode >= 400) {
+        return '4xx';
+    }
+
+    if (statusCode >= 300) {
+        return '3xx';
+    }
+
+    if (statusCode >= 200) {
+        return '2xx';
+    }
+
+    return '1xx';
+}
+
+class LifecycleMetrics {
```

Comment:
It would be wise to wrap all of these in try blocks, to avoid crashing the process in case anything (bug or runtime issue) throws, as this is unrelated to data flow.